### ISSUE TITLE
Typechecks in cmd interface

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#300]
This is something I discussed with @gtribello . We don't have time to do it now, but I paste some notes here for the future.

This issue originates from some points discussed in #289, where I proposed to add a cmd string to retrieve the list of cmds that accept a "non pointer" argument. Thinking in a more general way, we should probably have a procedure to interrogate plumed about the whole interface of all the cmd's strings. This would open the possibility for:
1. Find out automatically for which cmd you can allow conversion from data to pointer within the python wrapper.
2. Add some form of typechecking also from C++. It would be sufficient to include in the PLMD::Plumed object a map with allowed types and then overload the cmd() method (or use templates). One could have a runtime error if passing a pointer to an incorrect type.
3. Also from C++, allow conversion from data to pointer. I mean:
  `PLMD::Plumed p; p.cmd("setRealSize",8);`
would work as
  `PLMD::Plumed p; int r=8; p.cmd("setRealSize",&r);`

The kind of information needed could be provide as a single long string containing the specification of each cmd. This string could be parsed within the python/C++ constructor and used to construct a map later used for typechecking. It could be something like
````
cmd_setPositions (real*) lvalue
cmd_setRealSize (int*)
....
````

Here "real*" means that plumed expects a pointer it can later modify, where maybe lvalue says that plumed is storing the pointer, etc. The "real" type should be probably considered as special, in the sense that the wrapper should store the type of reals in order to check it in the following calls.

All these informations might be retrieved in the constructor, so as not to add overhead later. The information should be as simple as possible so that it can be parsed in the C++ or python constructor and in a format compatible across plumed versions, so we should choose it with care.

Also, since this will be (possibly) available in v2.5, one should check the APIversion before requesting the string to plumed and disable all the typechecks/conversions with plumed <=v2.4

I would say this should be done together with #182 (documentation of cmd strings).

**Comment by GiovanniBussi** [comment]
Too much work, so this is shifted to v2.6.

**Comment by GiovanniBussi** [comment]
closed in #653
